### ISSUE TITLE
Can we have a TODO List

### ISSUE THREAD

**Issue by VedantParanjape** [pine64/bl602-re#12]
Can there be a TODO List to get new people onboarded.

**Comment by Yangff** [comment]
I think there can be two things.. one is that we can rewrite these low-level libraries. This should be relatively easy since all symbols are preserved. 

Another thing is that we can reverse-engineering the reg table. Which seems to be hard, I already compared the regs from bl602 with bk7231. It turns out that they have different reg maps, and might even have different values for regs that shared same funcationality... at least from what I'm reading for `phy` and `modem`.

**Comment by VedantParanjape** [comment]
Hi, how can you rewrite low-level libraries without register map.

Can you link a WIP rewrite of any of these libraries just for reference.

**Comment by Yangff** [comment]
> Hi, how can you rewrite low-level libraries without register map.
> 
> Can you link a WIP rewrite of any of these libraries just for reference.

I' doing `phy_` functions now. That's easy because you know all the addresses and what's bing written to these addresses. The only thing you don't know is the meaning of them. But that doesn't stop you from writing exactly same code..

I assume there might be some legal problem, so the next step might be trying to figure out the reg map by playing with the values and see what happens to the hardware..

**Comment by VedantParanjape** [comment]
> ' doing phy_ functions now. That's easy because you know all the addresses and what's bing written to these addresses. The only thing you don't know is the meaning of them. But that doesn't stop you from writing exactly same code..

Makes sense, I'll take a look at this, and ask you any queries if that's okay

> I assume there might be some legal problem, so the next step might be trying to figure out the reg map by playing with the values and see what happens to the hardware..

I don't have the hardware yet :(

**Comment by gamelaster** [comment]
@VedantParanjape please, can you contact me at gamiee (at) pine64.dev ? Thank you 😊

**Comment by AshUK** [comment]
IMO a github project/issues board would be more appropriate as it would enable transparency and tracking.
